src/PIL/GimpPaletteFile.py

```diff
@@ -24,33 +25,49 @@ class GimpPaletteFile:
 
     rawmode = "RGB"
 
-    def __init__(self, fp):
+    #: override if reading larger palettes is needed
+    max_colors = 256
+    _max_line_size = 100
+    _max_file_size = 2**20
 
-        self.palette = [o8(i) * 3 for i in range(256)]
+    def __init__(self, fp):
 
         if fp.readline()[:12] != b"GIMP Palette":
             raise SyntaxError("not a GIMP palette file")
 
-        for i in range(256):
+        read = 0
 
-            s = fp.readline()
+        self.palette = b""
+        while len(self.palette) < 3 * self.max_colors:
+
+            s = fp.readline(self._max_file_size)
             if not s:
                 break
 
+            read += len(s)
+            if read >= self._max_file_size:
+                warnings.warn(
+                    f"Palette file truncated at {self._max_file_size - len(s)} bytes"
+                )
+                break
+
             # skip fields and comment lines
             if re.match(rb"\w+:|#", s):
                 continue
-            if len(s) > 100:
+            if len(s) > self._max_line_size:
                 raise SyntaxError("bad palette file")
 
-            v = tuple(map(int, s.split()[:3]))
-            if len(v) != 3:
+            # 4th column is color name and may contain spaces.
+            v = s.split(None, 4)
```

Comment:
```suggestion
            v = s.split(maxsplit=3)
```
3 is sufficient, as seen here -
```pycon
>>> "0 1 2 name".split(maxsplit=3)
['0', '1', '2', 'name']
```

Reply:
My bad. it is indeed the maximum "splits", not maximum "resulting components".